CMakeLists.txt

```diff
@@ -153,6 +153,12 @@ else()
   message(STATUS "Building without build tag")
 endif()
 
+find_package(Git)                                                                                            
+if(GIT_FOUND)
+  message(STATUS "Initializing submodules")
+  execute_process(COMMAND git "submodule" "update" "--init" "--recursive" WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
```

Comment:
--recursive is invasive and will pull in unnecessary submodules, such as rapidjson's gtest.